commit 6a241983a87544c905e664165de23834428c1fb3 (refs/changes/64/5564/1)
Author: Marsell Kukuljevic <marsell@joyent.com>
Date:   2019-02-11T22:00:57+01:00 (8 months ago)
    
    TRITON-1213 VM resize fails when tmpfs is 0 and no tmpfs set in package

diff --git a/lib/endpoints/vms.js b/lib/endpoints/vms.js
index c239556..1fdc466 100644
--- a/lib/endpoints/vms.js
+++ b/lib/endpoints/vms.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright (c) 2019, Joyent, Inc.
  */
 
 /*
@@ -706,6 +706,15 @@ function changeVm(req, res, next) {
         }
 
         req.log.debug({ params: params }, 'changeVm validated params');
+
+        // work around a mid-2018 to early-2019 VM.js platform bug, which causes
+        // resize jobs to fail on VMs with no tmpfs, even when the actual VM
+        // successfully resized
+        if (params.tmpfs == undefined && req.vm.tmpfs === 0) {
+            req.log.debug('changeVm set param.tmpfs to 0 since vm.tmpfs is 0');
+            params.tmpfs = 0;
+        }
+
         req.app.wfapi.createUpdateJob(req, params, function (err, juuid) {
             if (err) {
                 return next(err);
